### PR TITLE
Adding additional resources

### PR DESCRIPTION
**Pull request by kevgliss**
Hi, I am looking for some feedback on this PR.

I've added a few resources that we need to commonly resolve (security groups, vpc, subnets). 

Running the tests I get: 

`(node:45533) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot parse AWS type from ec2:test-name:testKey`

I'm not quite sure what this is trying to tell me, should I care? Should additional tests be added for these new resources?

### CODE REVIEW

**Comment by jacob-meacham** [comment]
@kevgliss thanks for the contribution!

That error is important; the plugin looks up the key via the AWS_HANDLERS map, so as written, instead of ec2:test-name:testKey, you'll want vpc:test-name:testKey. However, I think that your intuition is right -- I'd prefer to see a single handler for ec2, so you'd have ec2:vpc:vpcName and ec2:securityGroup:groupName. This will better match the aws API and so should hopefully be unsurprising.

Also, please do add tests, as we'd like to keep our test coverage high. Adding tests for the new resources should be fairly simple because the tests are relatively generic.

**Comment by kevgliss** [comment]
Cool, thanks for the insight. I will combine the above into a single handler.

**Comment by kevgliss** [comment]
I've updated this PR with the 'sub service' approach. It should not break existing functionality and should detect if a subservice is in use.  All of the tests pass and I've validated that it is working locally for me as well. 

I see that coverage decreased slightly. Let me know if that is an issue and I will attempt to add additional tests.

File: src/index.js [hunk 1]
```diff
@@ -22,6 +22,77 @@ async function getESSValue(key, awsParameters) {
 }
 
 /**
+ * @param key the name of the security group to resolve
+ * @param awsParameters parameters to pass to the AWS.EC2 constructor
+ * @returns {Promise<AWS.EC2.SecurityGroup>}
+ * @see http://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/EC2.html#describeSecurityGroups-property
+ */
+async function getEC2Value(key, awsParameters) {
+  winston.debug(`Resolving security group with name ${key}`)
```
Comment (jacob-meacham) [inline src/index.js]: Remove this print statement

**Comment by jacob-meacham** [comment]
Looks good, thanks for the PR!

**Comment by jacob-meacham** [comment]
Released as 1.1.0

**Pull request merged**